flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxExecutorImpl.java

```diff
@@ -68,14 +70,23 @@ public void executeFirst(
 
 	@Override
 	public void yield() throws InterruptedException {
-		mailbox.take(priority).run();
+		Mail mail = mailbox.take(priority);
+		try {
+			mail.run();
+		} catch (Exception ex) {
+			throw WrappingRuntimeException.wrapIfNecessary(ex);
+		}
 	}
```

Comment:
We execute user-supplied code here and call it e.g. from StreamTask.triggerCheckpointOnBarrier, so probably it's better not to hide the exception but to declare it or at least document with @ throws.

Reply:
But the designed use case of `yield` is inside user's operators. There we don't want to pollute user code with some random exceptions that he can not handle in any way (and we would force him to either declare `throws Exception` everywhere in his code, or to wrap it into the runtime anyway).